app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/WifiDirectManager.kt

```diff
@@ -275,13 +283,21 @@ class WifiDirectManager @Inject constructor(
         fileReceiverDeviceAddress =
           if (isGroupOwner) selectedPeerDeviceInetAddress else groupOwnerAddress
         activity.toast(R.string.preparing_files, Toast.LENGTH_LONG)
-        senderDeviceAsyncTask = SenderDeviceAsyncTask(this, activity).also {
-          it.execute(*filesForTransfer.toTypedArray())
+        senderDevice = SenderDevice(this, activity).also {
+          lifecycleCoroutineScope.launch {
+            val hasSend = it.send(filesForTransfer.toTypedArray())
```

Comment:
`hasSend` doesn't roll off the tongue. and we don't need to pass an array here, can just pass the list

Reply:
done. should we rename the `hasSend`?

Reply:
absolutely rename it

Reply:
done